tests/unit/intelligence/test_hatchet_worker_registration.py

```
@@ -327,11 +328,35 @@ async def test_run_agent_via_hatchet_regenerate_op_calls_regenerate(monkeypatch)
+    # node ids are normalized to NodeContext before reaching the service
+    assert args[1] == [NodeContext(node_id="n1", name="N")]
     assert args[2] == ["a1"]
     assert kwargs["run_id"] == "r1"
 
 
+async def test_run_agent_via_hatchet_normalizes_string_node_ids_for_message(
+    monkeypatch,
+):
+    """Plain string node ids from Hatchet payloads must be normalized to
+    NodeContext before MessageRequest is built, otherwise validation fails."""
+    sink, service, *_ = _install_fakes(monkeypatch)
```

Ruff (0.15.14)

[warning] Unpacked variable `sink` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)